### PR TITLE
chore: Upgrading dependencies

### PR DESCRIPTION
**Pull request by Tohaker**
Notable changes;

1. `@tsconfig/node18` upgraded from v1.0.1 to v18.2.0 to match their new versioning convention. This meant I had to change out some `module` compiler options as `commonjs` isn't needed anymore. Tested the UI and connections still work.
2. The new tsconfig meant that the Pothos types threw some errors, but the fix for this was to explicitly type Objects in the schema builder, nothing too untidy.

Other various upgrades include the build tools, type definitions, and some UI plugins. I ran through all the tests locally, and did a cursory manual check to make sure the UI still built and displayed as before.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,30 +11,30 @@
...
+    "eslint": "8.48.0",
     "eslint-config-prettier": "9.0.0",
-    "eslint-plugin-prettier": "5.0.0",
-    "eslint-plugin-react-hooks": "4.3.0"
+    "eslint-config-standard-with-typescript": "39.0.0",
+    "eslint-plugin-import": "2.28.1",
+    "eslint-plugin-n": "^16.0.0",
+    "eslint-plugin-prettier": "^5.0.0",
+    "eslint-plugin-promise": "^6.1.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
```
Comment (danstarns) [inline package.json:30]: I am curious to hear your thoughts @Tohaker on pinning vs. nonpinning dependencies in this project. 

Should we pin everything? Should we pin some and not others?

File: e2e/package.json [hunk 2]
```diff
@@ -23,7 +23,6 @@
     "@graphql-debugger/trace-schema": "workspace:^",
     "@graphql-debugger/data-access": "workspace:^",
     "puppeteer": "21.1.1",
-    "@types/puppeteer": "7.0.4",
```
Comment (danstarns) [inline e2e/package.json:26]: I noticed we removed this.

I get no build or IDE errors so they must now be included 😎

**Review by danstarns — APPROVED** [review]
This works well for me and in the pipelines - some big upgrades there 👌

Looking to adopt something in the pipelines for this so we can always keep up to date: 

https://github.com/rocket-connect/graphql-debugger/issues/23

File: e2e/package.json (hunk 2 above)
Comment (Tohaker) [reply e2e/package.json:26]: https://www.npmjs.com/package/@types/puppeteer Deprecated for a while now, so I removed them :)

File: package.json (hunk 1 above)
Comment (danstarns) [reply package.json:30]: With the addition of https://github.com/rocket-connect/graphql-debugger/issues/23 could we pin everything and then rely on that to apply patches etc.?

Comment (Tohaker) [reply package.json:30]: I pin `dependencies` and leave `devDependencies` unpinned. That way we control what's released to users but allow developers to have the patched versions of what they need to work with.

However, once we have an automated process bumping dependency versions whenever they're available, I'm comfortable to have _everything_ pinned.

Comment (danstarns) [reply package.json:30]: I like the idea of pinning the main ones and keeping the dev ones floating. 

Also, something to note here is that we have one expectation in the mono repo and that 

```
peerDependencies
    "graphql": "^16.0.0 || ^17.0.0"
```

In the utils package. 

Keen to hear thoughts on if this is the best approach to incude GraphQL in your project without clashing with other versions: 

```
Cannot use GraphQLSchema "[object Object]" from another module or realm.
```

Comment (Tohaker) [reply package.json:30]: I think that _should_ work in theory, although when I tried adding this one locally to my own existing (albeit far out of date) project it threw the error above. Might need to triage whether that's a setup issue we have or if it was just my crummy code 😆

**Pull request merged**